src/libYARP_dev/src/IVisualParamsImpl.cpp

```diff
@@ -202,12 +208,11 @@ bool Implement_RgbVisualParams_Parser::respond(const yarp::os::Bottle& cmd, yarp
                     ret = iRgbVisual->getRgbIntrinsicParam(params);
                     if(ret)
                     {
-                        yarp::os::Bottle params_b;
                         response.addVocab(VOCAB_RGB_VISUAL_PARAMS);
                         response.addVocab(VOCAB_INTRINSIC_PARAM);
                         response.addVocab(VOCAB_IS);
-                        ret &= Property::copyPortable(params, params_b);  // will it really work??
-                        response.append(params_b);
+                        yarp::os::Bottle& tmp=response.addList();
+                        ret &= Property::copyPortable(params, tmp);
```

Comment:
Yes, I understood your point.
It is not equivalent _in general_, while for boolean type it is since bool is (teoretically) 1 bit of information (altought not necessarily mapped on 1 bit on hw).
I'm just saying it is not wrong, and it is used all over yarp, so it looks strange to me you never saw it before.